Client/Frontend/Browser/FaviconManager.swift

```diff
@@ -38,7 +38,11 @@ class FaviconManager : BrowserHelper {
             if let icons = message.body as? [String: Int] {
                 for icon in icons {
                     if let iconUrl = NSURL(string: icon.0) {
-                        manager.downloadImageWithURL(iconUrl, options: SDWebImageOptions.LowPriority, progress: nil, completed: { (img, err, cacheType, success, url) -> Void in
+                        var options = [SDWebImageOptions.LowPriority]
+                        if browser?.isPrivate ?? false {
+                            options.append(SDWebImageOptions.CacheMemoryOnly)
+                        }
+                        manager.downloadImageWithURL(iconUrl, options: SDWebImageOptions(options), progress: nil, completed: { (img, err, cacheType, success, url) -> Void in
```

Comment:
If there's no `self.browser`, and we're a private tab, we might as well not run this code at all. So:

```
guard let browser = self.browser else {
    return
}
```